File: BiteClub/Models/Friend.swift
import Foundation

struct Friend: Codable, Identifiable {
    var id: UUID
    var userId: UUID
    var friendId: UUID
    var createdAt: Date
    var updatedAt: Date
    
    var user: User?
    var friend: User?
    
    enum CodingKeys: String, CodingKey {
        case id = "id"
        case userId = "user_id"
        case friendId = "friend_id"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
        case user = "user"
        case friend = "friend"
    }
}

File: BiteRunr/shared/src/commonMain/kotlin/org/biterunr_studios/biterunr/Models/Friend.kt
package org.biterunr_studios.biterunr.Models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.datetime.Instant

@Serializable
data class Friend(
    @SerialName("id")
    val id: String,

    @SerialName("user_id")
    val userId: String,

    @SerialName("friend_id")
    val friendId: String,

    @SerialName("created_at")
    val createdAt: Instant,

    @SerialName("updated_at")
    val updatedAt: Instant,

    @SerialName("user")
    val user: UserProfile? = null,

    @SerialName("friend")
    val friend: UserProfile? = null
)
